_posts/2023-10-23-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,311 @@
+Your custom resources will be the source-of-truth, like with the other operators.
+
+<!--more-->
+
+### Introduction
+
+Apache Kafka has emerged as a powerful and versatile platform for handling real-time data streams.
+It's used by organizations worldwide to ingest, process, and analyze data in a scalable and fault-tolerant manner.
+
+However, managing Kafka topics efficiently has often been a challenging task for administrators.
+These challenges often include creating topics, adjusting configurations, and ensuring consistency across the cluster, all while handling evolving requirements.
+Traditionally, these tasks have been performed manually, often leading to errors, inconsistencies, and operational overhead.
+
+This is where the Strimzi Topic Operator comes into play.
+It introduces a declarative approach to Kafka topic management, focusing on the "what" rather than the "how". 
+In a declarative model, administrators define the desired state of topics, specifying high-level characteristics and requirements, while leaving the intricate details to the operator itself.
+This shift in perspective not only simplifies the management process but also ensures that the desired configuration is maintained consistently across the Kafka cluster.
+
+<figure>
+    <img src="/assets/images/posts/2023-10-23-uto-interactions.png" height=440>
+    <figcaption><small>Fig 1. Topic Operator interactions.</small></figcaption>
+</figure>
+
+Declarative topic management involves defining the desired state of a Kafka topic using custom resources.
+In particular, you use Kubernetes custom resources to declare how you want your Kafka topics to be configured and managed.
+
+Here's how a Kafka topic is declared using this approach:
+
+```sh
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaTopic
+metadata:
+  name: my-topic
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  partitions: 3
+  replicas: 2
+  config:
+    retention.ms: 86400000
+```
+
+The requirement here is that a single `KafkaTopic` custom resource is used to manage a single topic in a single Kafka cluster.
+From the example above, you can see that the resource has a label `strimzi.io/cluster` with the name of the target cluster.
+The operator communicates with the Kafka cluster and ensures that the specified Kafka topic is created or updated according to the desired state.
+It can be deployed standalone, or along with your Kafka cluster using the `Kafka` custom resource.
+
+The first generation of the Topic Operator is based on the Kafka Streams library, and uses ZooKeeper to gather topic metadata.
+We call it **Bidirectional Topic Operator (BTO)**, as it allows to reconcile changes coming from both Kubernetes and Kafka using a three-way merge logic.
+In order to do this, it has to keep an internal topic store, which is the source-of-truth used to determine the necessary changes for synchronization.
```

Comment:
The last two sentences are a repetion of what you already said before. They are almost exactly the same sentences.

Reply:
Duplication removed.